src/webserver/DOSGuard.h

```diff
@@ -236,14 +615,75 @@ class BasicDOSGuard : public BaseDOSGuard
     [[nodiscard]] std::unordered_set<std::string> const
     getWhitelist(clio::Config const& config) const
     {
-        using T = std::unordered_set<std::string> const;
-        auto whitelist = config.arrayOr("dos_guard.whitelist", {});
+        // Data structures: string cast elements, complement, anwswer
```

Comment:
Since you have the function to check if it is a valid ip6/ip4 raw IP , if it is a valid ip6/ip4 subnet ip. I think you can use filter of copy_if to simply the loop logic. 
`